### PR TITLE
Added nRF monotonics

### PR DESCRIPTION
**Pull request by korken89**
Testing completed

### CODE REVIEW

File: rtic-monotonics/src/nrf/rtc.rs [hunk 1]
```diff
@@ -106,7 +106,10 @@ macro_rules! make_rtc {
 
                 $tq.initialize(Self {});
 
-                unsafe { pac::NVIC::unmask(Interrupt::$rtc) };
+                unsafe {
```
Comment (AfoHT) [inline rtic-monotonics/src/nrf/rtc.rs:115]: Maybe a good time to add an (un)safety

```
# Safety
blabla
```
comment (here and for other similar ones)?

**Review by perlindgren — APPROVED** [review]
Looks good.

bors merge

File: rtic-monotonics/src/nrf/rtc.rs [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+            type Instant = fugit::TimerInstantU64<32_768>;
+            type Duration = fugit::TimerDurationU64<32_768>;
+
+            fn now() -> Self::Instant {
+                // In a critical section to not get a race between overflow updates and reading it
+                // and the flag here.
+                critical_section::with(|_| {
+                    let rtc = unsafe { &*$rtc::PTR };
+                    let cnt = rtc.counter.read().bits() as u64;
+                    let ovf = if Self::is_overflow() {
```
Comment (Dirbaio) [inline rtic-monotonics/src/nrf/rtc.rs:193]: if overflow happens right between these lines, it'll give the wrong result

```rust
let cnt = rtc.counter.read().bits() as u64;
// OVERFLOW HAPPENS HERE
let ovf = if Self::is_overflow() {
```

Comment (perlindgren) [reply rtic-monotonics/src/nrf/rtc.rs:193]: So, a fix might be to re-read the `cnt` value on overflow.

**Pull request merged**